### PR TITLE
Added update data function

### LINKED ISSUE #235: Update data function
As a user, I should be able to update the information I stored in the database so I can see the updated version anytime.

### Acceptance
- [ ] I have a `updateOne` function that takes an Id and the details to update and makes calls to the zc_core UPDATE endpoint.

### PR DESCRIPTION
**Pull request by Augusta-E**
# What Does this PR do?
## This PR updates the plugin reminder 

- added database update function
- added controller update function
- added update route

### CODE REVIEW

File: src/controllers/reminder.controller.js [hunk 1]
```diff
@@ -150,6 +150,30 @@ const reminderController = {
...
+		const { priority, expiryDate, description, shouldRemind } = req.body
+
+		try {
+			const reminderData = { priority, expiryDate, description, shouldRemind }
+
+			const { id } = req.params
+
+			const updateReminder = await db.findByIdAndupdate(id, reminderData)
+
+			if (!id) {
```
Comment (iyosayi) [inline src/controllers/reminder.controller.js:181]: Why would you put this check here after updating? instead its like this

```javascript
if(!id) {}..
const updateReminder = await db.find...
```

Correct the order

**Review by iyosayi — CHANGES_REQUESTED** [review]
Implement those changes

File: src/controllers/reminder.controller.js (hunk 1 above)
Comment (Augusta-E) [reply src/controllers/reminder.controller.js:181]: oh! that was a mistake, let me quickly correct it

**Review by iyosayi — APPROVED** [review]
Looks good!

**Comment by iyosayi** [comment]
@Augusta-E Did you pull from `dev` branch first? Because your tests are failing. Please pull from dev branch and push again.

**Pull request merged**